### PR TITLE
Allow the daemon set to act on changes when an error occurs

### PR DESCRIPTION
**Pull request by KoishiKomeiji**
Previously, when the daemon set encounters an error during replication,
it would sleep for 5 minutes and then retry. It was not able to take in
updates or deletes until after the retry happens.

Now with this change it can take in updates in between this time period
and if the updates produce no error on replication it would cancel the
retry.

### CODE REVIEW

File: pkg/ds/daemon_set.go [hunk 1]
```diff
@@ -158,14 +163,23 @@ func (ds *daemonSet) WatchDesires(
 			if err != nil {
 				select {
 				case errCh <- err:
-					// Retry the replication for extra robustness
-					time.Sleep(RetryInterval)
-					err = ds.PublishToReplication()
+					ds.logger.Infof("An error has occurred in the daemon set, retrying if no changes are made in %v", RetryInterval)
```
Comment (anthonybishopric) [inline pkg/ds/daemon_set.go]: This should probably be a `Warnf`

File: pkg/ds/daemon_set.go [hunk 2]
```diff
@@ -158,14 +163,23 @@ func (ds *daemonSet) WatchDesires(
...
+					// Retry the replication in the RetryInterval's duration
+					timer.Reset(RetryInterval)
+					// This is required in case the user disables the daemon set
+					// so that the timer would be stopped after
+					err = nil
```
Comment (anthonybishopric) [inline pkg/ds/daemon_set.go]: Do we ever log the error?

Comment (KoishiKomeiji) [inline pkg/ds/daemon_set.go]: Yeah, it gets logged by the farm, the `Infof` a few lines above are a little more verbose in saying a retry will occur if nothing else happens.

**Comment by rudle** [comment]
:ship:

**Pull request merged**